examples/vision/super_resolution_sub_pixel.py

```diff
@@ -0,0 +1,471 @@
+"""
+Title: Image Super-Resolution Using an Efficient Sub-Pixel CNN
+Author: [Xingyu Long](https://github.com/xingyu-long)
+Date created: 2020/07/28
+Last modified: 2020/07/30
+Description: Implementing Super-Resoluion using Efficient sub-pixel model on BSDS500.
+"""
+
+"""
+## Introduction
+
+ESPCN (Efficient Sub-Pixel CNN), proposed by [Shi, W.2016](https://arxiv.org/abs/1609.05158)
+is used to reconstruct the image from low resolution (lowres) to high resolution (highres).
+In this paper, they introduce an efficient sub-pixel convolution layers which learns an array of
+upscaling filter to upscale the final lowres feature maps into the highres output. This post will
+implement the model in this paper and train it with small dataset
+[BSDS500](http://www.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/BSR/BSR_bsds500.tgz).
+"""
+
+"""
+## Setup
+"""
+
+import tensorflow as tf
```

Comment:
There are some imports throughout the notebook, prefer putting them all at the start of the file